packages/xrpl/src/Wallet/fundWallet.ts

```diff
@@ -190,31 +186,24 @@ async function processSuccessfulResponse(
       new XRPLFaucetError(`The faucet account is undefined`),
     )
   }
-  try {
-    // Check at regular interval if the address is enabled on the XRPL and funded
-    const updatedBalance = await getUpdatedBalance(
-      client,
-      classicAddress,
-      startingBalance,
-    )
+  // Check at regular interval if the address is enabled on the XRPL and funded
+  const updatedBalance = await getUpdatedBalance(
+    client,
+    classicAddress,
+    startingBalance,
+  )
 
-    if (updatedBalance > startingBalance) {
-      return {
-        wallet: walletToFund,
-        balance: updatedBalance,
-      }
+  if (updatedBalance > startingBalance) {
+    return {
+      wallet: walletToFund,
+      balance: updatedBalance,
     }
-    throw new XRPLFaucetError(
-      `Unable to fund address with faucet after waiting ${
-        INTERVAL_SECONDS * MAX_ATTEMPTS
-      } seconds`,
-    )
-  } catch (err) {
-    if (err instanceof Error) {
-      throw new XRPLFaucetError(err.message)
-    }
-    throw err
   }
+  throw new XRPLFaucetError(
+    `Unable to fund address with faucet after waiting ${
+      INTERVAL_SECONDS * MAX_ATTEMPTS
+    } seconds`,
+  )
```

Comment:
This error message is not relevant for the `processSuccessfulResponse` method. I do not see any retry-mechanism (or) back-off algorithm used here.

However, it is relevant for `getUpdatedBalance` method. These two error messages appear to have been swapped.

Reply:
No, the errors are correct. The faucet can also be used to add funds to an existing wallet, in which case the error should fall through to here, not throw in `getUpdatedBalance`.

Reply:
Can you point me to the code where we do the waiting (or) retry mechanism? Most of the usages of the faucet have been to generate a new wallet.

Even while funding an existing wallet, the operation is tried exactly once. I don't see the retry-operation.

Reply:
`setInterval` in `getUpdatedBalance`

Reply:
Yes, that's what I mentioned in my [first comment](https://github.com/XRPLF/xrpl.js/pull/2950#discussion_r2029453153). The `getUpdatedBalance` method is the correct method to place this error message, not `processSuccessfulResponse` method.

Reply:
`getUpdatedBalance` already has its own error message.